sktime/forecasting/conditional_invertible_neural_network.py

```diff
@@ -217,12 +229,20 @@ def _fit(self, y, fh, X=None):
             truncate="fill",
         ).fit_transform(y)
 
-        self.function = CurveFitForecaster(
-            self._f_statistic,
-            {"p0": self._init_param_f_statistic},
-            normalise_index=True,
-        )
-        self.function.fit(rolling_mean.dropna())
+        # Curve fit with error handling
+        try:
+            self.function = CurveFitForecaster(
+                self._f_statistic,
+                {"p0": self._init_param_f_statistic},
+                normalise_index=True,
+            )
+            self.function.fit(rolling_mean.dropna())
+        except Exception as e:
+            raise RuntimeError(
+                "Curve fitting error Please check the parameters and try again. "
+                f"Original error: {e}"
+            )
```

Comment:
```suggestion
            raise RuntimeError(
                "Curve fitting error. Please check the parameters and try again."
            ) from e
```

I suggest this, so that the original traceback is not lost.